File: 2023/06.23/줄어들지_않아.java
import java.io.*;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        int t = Integer.parseInt(br.readLine());
        long[][] dp = new long[65][10];

        for (int i = 0; i <= 9; i++) {
            dp[1][i] = 1;
        }

        for (int i = 2; i < dp.length; i++) {
            for (int j = 0; j < dp.length; j++) {
                for (int k = j; k <= 9; k++) {
                    dp[i][j] += dp[i-1][k];
                }
            }
        }

        for (int i = 0; i < t; i++) {
            int n = Integer.parseInt(br.readLine());
            long sum = 0;

            for (int j = 0; j <= 9; j++) {
                sum += dp[n][j];
            }

            bw.write(sum+"\n");
        }

        bw.flush();
        bw.close();
    }
}

File: 2023/06.23/줄어들지_않아.kt
import java.io.*

fun main() = with(BufferedReader(InputStreamReader(System.`in`))) {

    val bw = BufferedWriter(OutputStreamWriter(System.out))

    val t = readLine().toInt()
    val dp = Array(65){LongArray(10)}

    for (i in 0 .. 9){
        dp[1][i] = 1
    }

    for (i in 2 until dp.size){
        for (j in dp[i].indices){ // ex) dp[2][0]에는 0 ~ 9까지 들어갈 수 있으므로 += dp[i-1][k]에서 k가 0~9까지 들어갈 수 있다.
            for (k in j .. 9) // 해당 숫자부터 9까지 들어갈 수 있다.
            {
                dp[i][j] += dp[i-1][k]
            }
        }
    }

    repeat(t){
        bw.write("${dp[readLine().toInt() ].sum()}\n")
    }

    bw.flush()
    bw.close()
}